orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/MortService.java

```diff
@@ -0,0 +1,234 @@
+                String securityGroupName = (String) securityGroup.get("name");
+
+                if (securityGroupName == null) {
+                  List<SearchResult> searchResults =
+                      mortService.getSearchResults(
+                          (String) securityGroup.get("id"), "securityGroups");
+                  securityGroupName =
+                      searchResults != null
+                          ? (String) searchResults.get(0).results.get(0).get("name")
+                          : (String) securityGroup.get("id");
+                }
+
+                List<Map<String, Object>> portRanges =
+                    (List<Map<String, Object>>) inboundRule.get("portRanges");
+                final String finalSecurityGroupName = securityGroupName;
+                return portRanges.stream()
+                    .map(
+                        pr ->
+                            new SecurityGroupIngress(
+                                finalSecurityGroupName,
+                                (int) pr.get("startPort"),
+                                (int) pr.get("endPort"),
+                                (String) inboundRule.get("protocol")))
+                    .collect(Collectors.toList());
+              })
+          .flatMap(Collection::stream)
+          .collect(Collectors.toList());
+    }
+
+    static SecurityGroup findById(MortService mortService, String securityGroupId) {
+      List<SearchResult> searchResults =
+          mortService.getSearchResults(securityGroupId, "securityGroups");
+
+      Map securityGroup = null;
+      if (!searchResults.isEmpty()
+          && searchResults.get(0) != null
+          && searchResults.get(0).results != null
+          && searchResults.get(0).results.get(0) != null) {
+        securityGroup = searchResults.get(0).results.get(0);
+      }
+
+      if (securityGroup == null || securityGroup.get("name") == null) {
+        throw new IllegalArgumentException(
+            String.format("Security group (%s) does not exist", securityGroupId));
+      }
+
+      return mortService.getSecurityGroup(
+          (String) securityGroup.getOrDefault("account", null),
+          searchResults.get(0).platform,
+          (String) securityGroup.get("name"),
+          (String) securityGroup.getOrDefault("region", null),
+          (String) securityGroup.getOrDefault("vpcId", null));
+    }
+  }
+
+  class VPC {
+    String id;
+    String name;
+    String region;
+    String account;
+
+    static VPC findForRegionAndAccount(
+        Collection<VPC> allVPCs, String sourceVpcIdOrName, String region, String account) {
+      VPC sourceVpc =
+          allVPCs.stream()
+              .filter(
+                  vpc ->
+                      vpc.id.equalsIgnoreCase(sourceVpcIdOrName)
+                          || (sourceVpcIdOrName.equalsIgnoreCase(vpc.name)
+                              && vpc.region.equalsIgnoreCase(region)
```

Comment:
The groovy code for this is:
```
      def sourceVpc = allVPCs.find { it.id == sourceVpcIdOrName || (sourceVpcIdOrName.equalsIgnoreCase(it.name) && it.region == region && it.account == account) }
```
so aren't the comparisons for region and account case-sensitive...i.e.
```suggestion
                              && vpc.region.equals(region)
```

Reply:
Yep, not sure how I messed that up (and the ones below). Thanks!